third_party/move/move-vm/runtime/src/interpreter_caches.rs

```diff
@@ -13,13 +14,13 @@ use std::{
     sync::Arc,
 };
 
-/// Stable pointer identity for a [Function] within a single interpreter invocation.
+/// Stable pointer identity for a non-generic [Function] within a single interpreter invocation.
 #[derive(Copy, Clone, Eq, PartialEq, Debug)]
 pub(crate) struct FunctionPtr(*const Function);
 
 impl FunctionPtr {
     pub(crate) fn from_loaded_function(function: &LoadedFunction) -> Self {
-        FunctionPtr(Arc::as_ptr(&function.function))
+        Self(Arc::as_ptr(&function.function))
```

Comment:
Probably add a comment 'we can use pointer identity since the function loader guarantees that any loaded function has exactly one `Arc<Function>`'. 

Probably `FunctionPtr` declaration would be belong into the loader somewhere not here, because there is where this invariant is maintained.